backbone/src/main/java/org/researchstack/backbone/result/TappingIntervalResult.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    public Rect getButtonRect1() {
+        return buttonRect1;
+    }
+
+    public void setButtonRect1(Rect buttonRect1) {
+        this.buttonRect1 = buttonRect1;
+    }
+
+    public Rect getButtonRect2() {
+        return buttonRect2;
+    }
+
+    public void setButtonRect2(Rect buttonRect2) {
+        this.buttonRect2 = buttonRect2;
+    }
+
+    public static class Sample implements Serializable {
+        /**
+         * A relative timestamp indicating the time of the tap event.
+         *
+         * The timestamp is relative to the value of `startDate` in the `Result` object that includes this
+         * sample.
+         */
+        private long timestamp;
+
+        /**
+         * A duration of the tap event.
+         *
+         * The duration store time interval between touch down and touch release events.
+         */
+        private long duration;
+
+        /**
+         * An enumerated value that indicates which button was tapped, if any.
+         *
+         * If the value of this property is `ORKTappingButtonIdentifierNone`, it indicates that the tap
+         * was near, but not inside, one of the target buttons.
+         */
+        private TappingButtonIdentifier buttonIdentifier;
+
+        /**
+         * The location of the tap within the step's view.
+         *
+         * The location coordinates are relative to a rectangle whose size corresponds to
+         * the `stepViewSize` in the enclosing `ORKTappingIntervalResult` object.
+         */
+        private Point location;
+
+        /* Default identifier for serialization/deserialization */
+        public Sample() {
+            super();
+        }
+
+        public long getTimestamp() {
+            return timestamp;
+        }
+
+        public void setTimestamp(long timestamp) {
+            this.timestamp = timestamp;
+        }
+
+        public long getDuration() {
+            return duration;
+        }
+
+        public void setDuration(long duration) {
+            this.duration = duration;
+        }
+
+        public TappingButtonIdentifier getButtonIdentifier() {
+            return buttonIdentifier;
+        }
+
+        public void setButtonIdentifier(TappingButtonIdentifier buttonIdentifier) {
+            this.buttonIdentifier = buttonIdentifier;
+        }
+
+        public Point getLocation() {
+            return location;
+        }
+
+        public void setLocation(Point location) {
+            this.location = location;
+        }
+    }
+
+    /**
+     Values that identify the button that was tapped in a tapping sample.
+     */
+    public enum TappingButtonIdentifier {
+        // The touch landed outside of the two buttons.
+        TappingButtonIdentifierNone,
+        // The touch landed in the left button.
+        TappingButtonIdentifierLeft,
+        // The touch landed in the right button.
+        TappingButtonIdentifierRight;
+    }
+
+    public static final class Size implements Serializable {
```

Comment:
Yeah, they aren't serializable.  You also can't override them and make them serializable because they are final classes.